app/scripts/models/account.js

```diff
@@ -802,9 +802,16 @@ define(function (require, exports, module) {
             return profileClient[method].apply(profileClient, [accessToken].concat(args));
           })
           .fail(function (err) {
-            // If no oauth token existed, or it has gone stale,
-            // get a new one and retry.
-            if (ProfileClient.Errors.is(err, 'UNAUTHORIZED')) {
+            if (ProfileClient.Errors.is(err, 'INVALID_TOKEN')) {
+              // invalid token can happen if user uses 'Disconnect' in Firefox Desktop.
+              // Only 'set' will trigger model change, using 'unset' will not.
+              // Details: github.com/jashkenas/backbone/issues/949
+              self.set('accessToken', null);
```

Comment:
That's strange, the [Backbone docs say](http://backbonejs.org/#Model-unset) unset `Fires a "change" event unless silent is passed as an option.`